pkg/server/server.go

```diff
@@ -0,0 +1,629 @@
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments").Methods("POST").HandlerFunc(createEnvironment)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}").Methods("GET").HandlerFunc(getEnvironment)
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}").Methods("PATCH").HandlerFunc(updateEnvironment)
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}").Methods("DELETE").HandlerFunc(deleteEnvironment)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments").Methods("GET").HandlerFunc(getDeployments)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments").Methods("POST").HandlerFunc(createDeployment)
+
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments/{deployment}").Methods("GET").HandlerFunc(getDeployment)
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments/{deployment}").Methods("PATCH").HandlerFunc(updateDeployment)
+	sub.Path("/environmentGroups/{environmentGroupID}/environments/{environment}/deployments/{deployment}").Methods("DELETE").HandlerFunc(deleteDeployment)
+
+	server = &Server{
+		Router: router,
+	}
+	return server
+}
+
+//Start the server
+func (server *Server) Start() error {
+	return http.ListenAndServe(":9000", server.Router)
+}
+
+//Route handlers
+
+//getEnvironmentGrouos returns a list of all Environment Groups
+//What is an environmentGroup?
+func getEnvironmentGroups(w http.ResponseWriter, r *http.Request) {
+	//TODO: What is this supposed to do?
+	//For now I'll just return 404 I guess...
+	http.Error(w, "404 Page not found", 404)
+}
+
+//getEnvironmentGroup returns an Environment Group matching the given environmentGroupID
+//What is an environmentGroup?
+func getEnvironmentGroup(w http.ResponseWriter, r *http.Request) {
+	// vars := mux.Vars(r)
+
+	//TODO: What is this supposed to do?
+	//For now I'll just return 404 I guess...
+	http.Error(w, "404 page not found", 404)
+}
+
+//getEnvironments returns a list of all environments under a specific environmentGroupID
+func getEnvironments(w http.ResponseWriter, r *http.Request) {
+	pathVars := mux.Vars(r)
+
+	selector, err := labels.Parse("Group=" + pathVars["environmentGroupID"])
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error creating label selector: %v\n", err)
+		return
+	}
+	nsList, err := client.Namespaces().List(api.ListOptions{
+		LabelSelector: selector,
+	})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error in getEnvironments: %v\n", err)
+		fmt.Fprintf(w, "%v\n", err)
+		return
+	}
+	js, err := json.Marshal(nsList)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		fmt.Printf("Error marshalling namespace list: %v\n", err)
+	}
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
I see this used in a number of places.  You could put this in a helper or something where you set the necessary header(s) and marshal to a JSON string.  Not required for this but just a thought going forward.